File: connection-examples/typescript/src/insert.ts
// node 14+ cjs named exports not found in pg
import pkg from 'pg';
const { Client } = pkg;

const client = new Client({
    user: "MATERIALIZE_USERNAME",
    database: "materialize",
    password: "APP_SPECIFIC_PASSWORD",
    host: "MATERIALIZE_HOST",
    port: 6875,
    ssl: true
});

// const createTable = 'CREATE TABLE countries (code text NOT NULL, name text NOT NULL);';
const text = 'INSERT INTO countries(code, name) VALUES($1, $2);';
const values = ['GH', 'GHANA'];

async function main() {
    try {
        await client.connect();
        const res = await client.query(text, values);
        console.log(res);
    } catch(e) {
        console.log("error: ", e);
    } finally {
        await client.end();
    }
}

await main();

File: connection-examples/rust/src/insert.rs
use postgres::{Error};

use crate::connection::create_client;

/// Insert data into the table
pub(crate) fn insert() -> Result<u64, Error> {
    let mut client = create_client().expect("Error creating client.");

    let code = "GH";
    let name = "Ghana";

    client.execute("INSERT INTO countries(code, name) VALUES($1, $2)", &[&code, &name])
}